src/Stripe.net/Infrastructure/Public/StripeConfiguration.cs

```diff
@@ -176,6 +180,30 @@ public static JsonSerializerSettings DefaultSerializerSettings()
             };
         }
 
+        /// <summary>
+        /// Sets information about your application. The information is passed along to Stripe.
+        /// </summary>
+        /// <param name="name">Name of your application (e.g. "MyAwesomeApp")</param>
+        /// <param name="partnerId">Your Stripe Partner ID (e.g. "pp_partner_1234")</param>
+        /// <param name="version">Version of your application (e.g. "1.2.34")</param>
+        /// <param name="url">Website for your application (e.g. "https://myawesomeapp.info")</param>
+        public static void SetAppInfo(
+            string name,
+            string partnerId = null,
+            string version = null,
+            string url = null)
+        {
+            AppInfo = new AppInfo
+            {
+                Name = name,
+                PartnerId = partnerId,
+                Version = version,
+                Url = url,
+            };
+
+            ((StripeClient as StripeClient)?.HttpClient as SystemNetHttpClient)?.RebuildUserAgentStrings();
```

Comment:
This is pretty ugly, but I really don't want to change the `IStripeClient` and `IHttpClient` interfaces just to add support for this.

Reply:
Yeah I don't understand that line. Mind asking someone else? :p